pkg/live/inventoryrg.go

```diff
@@ -249,78 +308,432 @@ func (icm *InventoryResourceGroup) GetObject() (*unstructured.Unstructured, erro
+	}
+
+	// Delete ResourceGroups that contain duplicated GKNNs.
+	for rgIdx := len(resourcegroups) - 1; rgIdx >= objIdx/ResourceCount; rgIdx-- {
+		err = namespacedClient.Delete(context.TODO(), resourcegroups[rgIdx].GetName(), metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// createResourceGroups creates new ResourceGroup object(s) on cluster, and handles the sharding of managed
+// resources into multiple ResourceGroup objects.
+func (icm *InventoryResourceGroup) createResourceGroups(dc dynamic.Interface, mapper meta.RESTMapper) error {
+	namespacedClient, err := icm.getNamespacedClient(dc, mapper)
+	if err != nil {
+		return err
+	}
+
+	// totalStored tracks the total number of resources stored within all sharded ResourceGroups.
+	totalStored := 0
+
+	for totalStored < len(icm.objMetas) {
+		var items []interface{}
+
+		switch ResourceCount {
+		case DisableSharding:
+			for _, objMeta := range icm.objMetas {
+				items = append(items, idToUnstructuedResources(objMeta))
+			}
+		default: // Split managed objects into current ResourceGroup.
+			for partitionCount := 0; partitionCount < ResourceCount && totalStored+partitionCount < len(icm.objMetas); partitionCount++ {
+				objMeta := icm.objMetas[totalStored+partitionCount]
+				items = append(items, idToUnstructuedResources(objMeta))
+			}
+		}
+
+		var rg *unstructured.Unstructured
+		switch totalStored {
+		case 0: // Initial RG in series.
+			rg = icm.inv
+			rg.SetResourceVersion("")
+		default:
+			rg = icm.inv.DeepCopy()
+			rg.SetResourceVersion("")
+			rg.SetName(fmt.Sprintf(shardedRGNameFmt, rg.GetName(), len(rg.GetName()), totalStored/ResourceCount))
+		}
+
+		addShardedLabel(rg, icm.Name())
+
+		err := unstructured.SetNestedSlice(rg.Object, items, "spec", "resources")
+		if err != nil {
+			return err
+		}
+
+		_, err = namespacedClient.Create(context.TODO(), rg, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+
+		totalStored += len(items)
+	}
+
+	return nil
 }
 
-func (icm *InventoryResourceGroup) getNamespacedClient(dc dynamic.Interface, mapper meta.RESTMapper) (*unstructured.Unstructured, dynamic.ResourceInterface, error) {
+// getNamespacedClient returns a namespaced client for interacting with ResourceGroups on cluster.
+func (icm *InventoryResourceGroup) getNamespacedClient(dc dynamic.Interface, mapper meta.RESTMapper) (dynamic.ResourceInterface, error) {
 	invInfo, err := icm.GetObject()
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	if invInfo == nil {
-		return nil, nil, fmt.Errorf("attempting to create a nil inventory object")
+		return nil, errors.New("attempting to interact with a nil inventory object")
 	}
 
 	mapping, err := mapper.RESTMapping(invInfo.GroupVersionKind().GroupKind(), invInfo.GroupVersionKind().Version)
 	if err != nil {
-		return nil, nil, err
+		return nil, fmt.Errorf("unable to create rest mapper for %s ResourceGroup: %w", invInfo.GroupVersionKind().Version, err)
```

Comment:
Inaccurate error message. It couldn't create a REST "mapping" for a specific resource, not a "mapper". The mapper is the struct that resolves mappings.

Also, I suspect the RESTMapping includes the GVK in the error message, but you could check. You're only adding the version here, which seems weird.